nucleus/dataset.py

```diff
@@ -78,12 +95,28 @@ class Dataset:
         existing_dataset = client.get_dataset("ds_bwkezj6g5c4g05gqp1eg")
     """
 
-    def __init__(self, dataset_id, client):
+    def __init__(
+        self,
+        dataset_id: str,
+        client: "NucleusClient",  # noqa: F821
+        name: Optional[str] = None,
+    ):
+        """
+
+        Args:
+            dataset_id: Reference id of dataset
+            client: NucleusClient instance
+            name: Optionally set name on creation such that the property access doesn't need to hit the server
+        """
         self.id = dataset_id
         self._client = client
+        self._name = name
```

Comment:
For now we can add your comment here
```suggestion
        self._name = name # Optionally set name on creation such that the property access doesn't need to hit the server
```

Reply:
Nice, thanks!